packages/app/src/cli/services/bulk-operations/execute-bulk-operation.ts

```diff
@@ -64,31 +67,45 @@ export async function executeBulkOperation(input: ExecuteBulkOperationInput): Pr
     return
   }
 
-  const result = bulkOperationResponse?.bulkOperation
-  if (result) {
-    const infoSections = [
-      {
-        title: 'Bulk Operation Created',
-        body: [
-          {
-            list: {
-              items: [
-                outputContent`ID: ${outputToken.cyan(result.id)}`.value,
-                outputContent`Status: ${outputToken.yellow(result.status)}`.value,
-                outputContent`Created: ${outputToken.gray(String(result.createdAt))}`.value,
-              ],
-            },
-          },
-        ],
-      },
-    ]
-
-    renderInfo({customSections: infoSections})
-
-    renderSuccess({
-      headline: 'Bulk operation started successfully!',
-      body: 'Congrats!',
-    })
+  const createdOperation = bulkOperationResponse?.bulkOperation
+  if (createdOperation) {
+    if (watch) {
+      const finishedOperation = await watchBulkOperation(adminSession, createdOperation.id)
+      renderBulkOperationResult(finishedOperation)
+    } else {
+      renderBulkOperationResult(createdOperation)
+    }
+  }
+}
+
+function renderBulkOperationResult(operation: BulkOperation): void {
+  const headline = formatBulkOperationStatus(operation).value
+  const items = [
+    outputContent`ID: ${outputToken.cyan(operation.id)}`.value,
+    outputContent`Status: ${outputToken.yellow(operation.status)}`.value,
+    outputContent`Created at: ${outputToken.gray(String(operation.createdAt))}`.value,
+    ...(operation.completedAt
+      ? [outputContent`Completed at: ${outputToken.gray(String(operation.completedAt))}`.value]
+      : []),
+  ]
+
+  const customSections = [{body: [{list: {items}}]}]
+
+  switch (operation.status) {
+    case 'CREATED':
+      renderSuccess({headline: 'Bulk operation started.', customSections})
+      break
+    case 'COMPLETED':
+      if (operation.url) {
+        const downloadMessage = outputContent`Download results ${outputToken.link('here.', operation.url)}`.value
+        renderSuccess({headline, body: [downloadMessage], customSections})
+      } else {
+        renderSuccess({headline, customSections})
```

Comment:
Maybe I'm missing something, but when there is no URL, what are you supposed to do next?

```
Bulk operation started.

• ID: gid://shopify/BulkOperation/5717783019762
• Status: CREATED
• Created at: 2025-11-26T08:51:33Z
```

I'd show some help about how to get the results. If they need to run another query, couldn't we generate them so they can copy and paste? Or print a URL they can check later?